discovery-infra/test_infra/controllers/proxy_controller/proxy_controller.py

```diff
@@ -0,0 +1,72 @@
+import os
+import shutil
+import socket
+import logging
+from jinja2 import Environment, PackageLoader
+
+PROXY_USER = 'assisted'
+PROXY_USER_PASS = 'redhat'
+PODMAN_FLAGS = '--cgroup-manager=cgroupfs --storage-driver=vfs --events-backend=file'
+
+class ProxyController():
+    def __init__(self, name=None, port=None, denied_port=None, authenticated=False):
+        if not name or not port:
+            self.address = ''
+        else:
+            self.name = name
+            self.port = port
+            self.authenticated = authenticated
+            self._pull_image()
+            self._set_server_address()
+            self._create_conf_from_template(denied_port=denied_port)
+            self._create_user_file_for_auth()
+            self._run_proxy_server()
+
+    def remove(self):
+        if self.address:
+            logging.info(f'Removing Proxy Server {self.name}')
+            container_rm_cmd = f'podman stop {self.name} {PODMAN_FLAGS} && podman rm {self.name} {PODMAN_FLAGS}'
+            os.system(container_rm_cmd)
+
+            self._remove_config()
+
+    def _pull_image(self):
```

Comment:
We will probably need this function (and few others) to be more generic,for example, in NTP ill need to create a new container, ill be glad to use those functions to raise up new container.
Another controller maybe?

Reply:
Sure, I'll add a container_controller that proxy_controller will inherit from

Reply:
On second thought, ContainerController is overkill and clumsy
Moving relevant methods to utils

Reply:
Added pull_container_image, run_container, and remove_running_container to utils

Reply:
changed